packages/monaco/src/browser/monaco-command.ts

```diff
@@ -120,6 +127,93 @@ export class MonacoEditorCommandHandlers implements CommandContribution {
             }
         };
     }
+    protected newConfigIndentationHandler(): MonacoEditorCommandHandler {
+        return {
+            execute: (editor: MonacoEditor) => {
+                const configTabSize = (useSpaces: boolean) => {
+                    const editorModel = editor.document;
+                    if (editorModel && editorModel.textEditorModel) {
+                        const tabSize = editorModel.textEditorModel.getOptions().tabSize;
+                        const configuredTabSize = this.editorPreferences['editor.tabSize'];
+                        const sizes = Array.from(Array(8), (_, x) => x + 1);
+                        const tabSizeOptions = sizes.map(size =>
+                            new UpdateTabSizeOpenItem(size === configuredTabSize ? `${size}   Configured Tab Size` : size.toString(),
+                                useSpaces,
+                                () => {
+                                    editorModel.textEditorModel.updateOptions({
+                                        tabSize: size || tabSize,
+                                        insertSpaces: useSpaces
+                                    });
+                                },
+                                (mode: QuickOpenMode) => mode === QuickOpenMode.OPEN,
+                                () => true
+                            ));
+                        this.open(tabSizeOptions,
+                            'Select Tab Size for Current File',
+                            (lookFor: string) => {
+                                if (!lookFor || lookFor === '') {
+                                    return tabSize - 1;
+                                }
+                                return sizes.findIndex(s => lookFor.startsWith(s.toString()));
+                            });
+                    }
+                };
+                const options: QuickOpenItem[] = [true, false].map(useSpaces =>
+                    new UpdateTabSizeOpenItem(`Indent Using ${useSpaces ? 'Spaces' : 'Tab Size'}`,
+                        useSpaces,
+                        configTabSize,
+                        (mode: QuickOpenMode) => mode === QuickOpenMode.OPEN,
+                        () => false)
+                );
+                this.open(options, 'Select Action');
+            }
+        };
+    }
+    protected newConfigTabSizeHandler(useSpaces: boolean): MonacoEditorCommandHandler {
+        return {
+            execute: (editor: MonacoEditor) => {
+                const editorModel = editor.document;
+                const tabSize = editorModel.textEditorModel.getOptions().tabSize;
+                const configuredTabSize = this.editorPreferences['editor.tabSize'];
+                const sizes = Array.from(Array(8), (_, x) => x + 1);
+                const tabSizeOptions = sizes.map(size =>
+                    new UpdateTabSizeOpenItem(size === configuredTabSize ? `${size}   Configured Tab Size` : size.toString(),
```

Comment:
I don't think you need `UpdateTabSizeOpenItem`, you can achieve the same with `QuickOpenItem` and options:
```ts
new QuickOpenItem({
    label: size === configuredTabSize ? `${size}   Configured Tab Size` : size.toString(),
    execute: mode => {
        if (mode !== QuickOpenMode.OPEN) {
            return false;
        }
        editorModel.textEditorModel.updateOptions({
            tabSize: size || tabSize,
            insertSpaces: useSpaces
        });
        return true;
    }
})
```